### PR TITLE
Move quarkus-devtools-utils and the codestarts artifact to the quarkus-bom

### PR DESCRIPTION
**Pull request by aloubyansky**
These are used by some core artifacts.

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -179,22 +179,6 @@
...
-            </dependency>
-            <dependency>
-                <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-project-core-extension-codestarts</artifactId>
-                <version>${project.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-project-core-extension-codestarts</artifactId>
-                <type>pom</type>
```
Comment (gastaldi) [inline build-parent/pom.xml:195]: Isn't this meant to be moved too?

Comment (aloubyansky) [reply build-parent/pom.xml:195]: I didn't move it on purpose. Including POMs in the user BOMs is generally useless.

File: devtools/gradle/gradle-application-plugin/pom.xml [hunk 2]
```diff
@@ -37,6 +37,7 @@
             <artifactId>quarkus-project-core-extension-codestarts</artifactId>
             <type>pom</type>
             <scope>test</scope>
+            <version>${project.version}</version>
```
Comment (gastaldi) [inline devtools/gradle/gradle-application-plugin/pom.xml:40]: Isn't better to keep this specific artifact in build-parent instead?

Comment (aloubyansky) [reply devtools/gradle/gradle-application-plugin/pom.xml:40]: We could. But then we could do the same for the rest of dependencies on POMs, e.g. https://github.com/quarkusio/quarkus/blob/main/integration-tests/amazon-lambda-http-resteasy-reactive/pom.xml#L40

**Review by gastaldi — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]
Thanks @aloubyansky!

**Comment by ppalaga** [comment]
Could we please get this backported to 2.7?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@ppalaga yep, it already has the label.

**Pull request merged**